### DISCUSSION TITLE
Help understanding use of the refresh-token

### DISCUSSION THREAD

**Discussion by tomelliot** [supabase/supabase#17576] [Questions]
Hi there,

I'm looking for some more information on how to treat the `access-token` and `refresh-token`. As the `refresh-token` is single-use, if I use it to `supabase.auth.setSession` in another context, does that impact the original session, or is there any conflict between these two sessions?

More details on my use case:
I am building an SPA that runs on a 3rd party domain as a plugin/extension. I'd like a user to be automatically logged into the plugin, if they're logged in on my domain.

The solution I have so far:
- When user logins in to `mydomain.com`, create cross-site cookies with the `auth-token` and `refresh-token`. (needs to be cross-site, as the iframe is not considered to be SameSite as `mydomain.com`)
- In the SPA embed an iframe of `mydomain.com/plugin-login`.
- From the iframe, use `window.parent.postMessage` to pass the two tokens.
- In the SPA capture the tokens from the iframe and `supabase.auth.setSession({access_token: access_token, refresh_token: refresh_token})`

This seems to be working, but I'd like to make sure I'm not going to end up with the user's session being expired due to the `refresh_token` already being used.

**Comment by GaryAustin1** [comment]
I believe if the session is expired or expiring then it will burn your token.
![image](https://github.com/supabase/supabase/assets/54564956/d2ef5127-93a3-4232-8783-1518b898c8be)

**Comment by tomelliot** [comment]
Thanks @GaryAustin1 - that makes sense, so the way I'm understanding that: If I use the `refresh-token` on the SPA, then the `mydomain.com` session will not be able to renew.

Is there a way to achieve what I'm aiming for here?

**Reply by robertsamarji** [reply]
Hey @tomelliot, did you find a solution to this?

**Reply by tomelliot** [reply]
@robertsamarji No - I've moved to forcing the user to login in both locations. If you come up with a better solution please let us know!
